### PR TITLE
Add instructions for developing mongochemclient

### PR DESCRIPTION
**Pull request by psavery**

### CODE REVIEW

**Comment by psavery** [comment]
@cjh1 @cryos I added a section to the dev/README.md titled `Developing MongoChemClient`. Let me know if any changes are needed.

File: dev/nginx/oc_client [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	# Read up on ssl_ciphers to ensure a secure configuration.
+	# See: https://bugs.debian.org/765782
+	#
+	# Self signed certs generated by the ssl-cert package
+	# Don't use them in a production server!
+	#
+	# include snippets/snakeoil.conf;
+
+	root /var/www/html;
+        #root /home/alessandro/kitware/mongochemclient_webcomponent/build;
```
Comment (cjh1) [inline dev/nginx/oc_client]: Let remove this!

Comment (cjh1) [reply dev/nginx/oc_client]: In fact let remove all the commented out config ....

Comment (cjh1) [reply dev/nginx/oc_client]: And lets not use 80, lets use 9090?

File: dev/README.md [hunk 2]
```diff
@@ -37,6 +37,29 @@ container.
 
 Once ```<repo>/dev/dev.env``` has been update you can bring the stack back up using ```ocdeploy up```.
 
+Developing MongoChemClient
+--------------------------
+
+In order to visualize updates to the client before publishing, some extra steps may be needed.
+
+First, as mentioned in the section above, ```MONGOCHEMCLIENT``` in ```<repo>/dev/dev.env``` must
```
Comment (alesgenova) [inline dev/README.md]: This step is not needed if you're going for the nginx settings route

**Comment by psavery** [comment]
I removed some configurations. Let me know if I should remove more.

Also, I just realized that I should have probably used a different branch name... not master.

File: dev/README.md (hunk 2 above)
Comment (psavery) [reply dev/README.md]: Okay, I got rid of that step.

File: dev/nginx/oc_client (hunk 1 above)
Comment (psavery) [reply dev/nginx/oc_client]: Done and done. I used 9090.

**Comment by psavery** [comment]
@cjh1 When I changed the oc_client to use localhost:9090 instead of localhost:80, I started running into an issue loading the notebooks. See screenshot below. Maybe I need to modify something else also?

Screenshot from 2019-03-28 13-06-35

**Comment by psavery** [comment]
If I try to click the `Python 3` under `Notebook`, I get this error as well.
Screenshot from 2019-03-28 13-14-09

File: dev/nginx/oc_client [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        proxy_http_version 1.1;
+        chunked_transfer_encoding off;
+        proxy_read_timeout 600s;
+        proxy_send_timeout 600s;
+        }
+
+         location /jupyterhub/ {
+         proxy_pass http://localhost:8000/jupyterhub/;
+         proxy_set_header X-Real-IP $remote_addr;
+         proxy_set_header Host $host;
```
Comment (cjh1) [inline dev/nginx/oc_client]: You would need :9090 here.

File: dev/nginx/oc_client [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+         proxy_set_header Upgrade $http_upgrade;
+         proxy_set_header Connection $connection_upgrade;
+         }
+
+
+	location / {
+		# First attempt to serve request as file, then
+		# as directory, then fall back to displaying a 404.
+                        proxy_redirect      off;
+        proxy_set_header    Host $host;
```
Comment (cjh1) [inline dev/nginx/oc_client]: Here as well

**Comment by psavery** [comment]
I think this is now working, as far as I can tell. Thanks Chris!

**Comment by psavery** [comment]
@cjh1 I have been using the new `oc_client` file, and I think it has been working fine.

Does this look good to merge?

File: dev/nginx/oc_client [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+	# Note: You should disable gzip for SSL traffic.
+	# See: https://bugs.debian.org/773332
+	#
+	# Read up on ssl_ciphers to ensure a secure configuration.
+	# See: https://bugs.debian.org/765782
+	#
+	# Self signed certs generated by the ssl-cert package
+	# Don't use them in a production server!
+	#
+	# include snippets/snakeoil.conf;
```
Comment (cjh1) [inline dev/nginx/oc_client]: Let remove this commented out block

File: dev/nginx/oc_client [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+	#	# With php-cgi (or other tcp sockets):
+	#	fastcgi_pass 127.0.0.1:9000;
+	#}
+
+	# deny access to .htaccess files, if Apache's document root
+	# concurs with nginx's one
+	#
+	#location ~ /\.ht {
+	#	deny all;
+	#}
```
Comment (cjh1) [inline dev/nginx/oc_client]: This block as well

File: dev/nginx/oc_client [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+#
+#	server_name example.com;
+#
+#	root /var/www/example.com;
+#	index index.html;
+#
+#	location / {
+#		try_files $uri $uri/ =404;
+#	}
+#}
```
Comment (cjh1) [inline dev/nginx/oc_client]: All this a well

**Comment by psavery** [comment]
Done.

**Pull request merged**